Dockerfile

```diff
@@ -0,0 +1,42 @@
+# syntax=docker/dockerfile:1.4
+
+ARG PYTHON_VERSION="3.11.4"
+FROM python:${PYTHON_VERSION}-slim-bullseye
+
+SHELL ["/bin/bash", "-o", "pipefail", "-e", "-u", "-x", "-c"]
+
+# Setup Pip
+ARG PIP_VERSION="23.2.1"
+ENV PIP_VERSION=${PIP_VERSION}
+
+RUN pip install --no-cache-dir --upgrade "pip==${PIP_VERSION}" && pip --version
+
+ENV PYTHONUNBUFFERED=1
+ENV PIP_ROOT_USER_ACTION=ignore
+RUN mkdir /component
+WORKDIR /component
+
+# Install streamlit and components
+ARG STREAMLIT_VERSION="latest"
+ENV E2E_STREAMLIT_VERSION=${STREAMLIT_VERSION}
+
+# hadolint ignore=DL3013
```

Comment:
Do we have hadolint setup for this repo?

Reply:
Copy-pasted :P I am removing that comment unless you think that we need hadolint. As for me, I think we can live without it

Reply:
Can you create an issue in EPIC: https://snowflakecomputing.atlassian.net/browse/STREAMLIT-3913 ?